kasa/smartprotocol.py

```diff
@@ -157,15 +147,50 @@ async def _execute_query(self, request: Union[str, Dict], retry_count: int) -> D
             self._host,
             _LOGGER.isEnabledFor(logging.DEBUG) and pf(smart_request),
         )
-        response_data = await self._transport.send(smart_request)
+        resp_dict = await self._transport.send(smart_request)
 
         _LOGGER.debug(
             "%s << %s",
             self._host,
-            _LOGGER.isEnabledFor(logging.DEBUG) and pf(response_data),
+            _LOGGER.isEnabledFor(logging.DEBUG) and pf(resp_dict),
         )
 
-        return response_data
+        self._handle_response_error_code(resp_dict)
+
+        if result := resp_dict.get("result"):
+            if is_multi:
+                if (responses := result.get("responses")) is None:
+                    raise SmartDeviceException(
+                        "Unexpected response to multipleRequest "
+                        + f"method for {self._host}: {result}"
+                    )
```

Comment:
Maybe we could get rid off the `is_multi` helper variable, and just check for the existence of `responses` property? The only downside for making the code more complex (i.e., having more branches to write test code) is that we could spot some potential firmware bugs?

Reply:
Done